GameServer/Logic/Maps/SummonersRift.cs

```diff
@@ -349,37 +419,91 @@ public override bool Spawn()
                 MinionSpawnPosition.SPAWN_RED_BOT,
                 MinionSpawnPosition.SPAWN_RED_MID,
             };
-
-            if (_waveNumber < 3)
+            foreach (var pos in positions)
             {
-                for (var i = 0; i < positions.Count; ++i)
+                uint inhibitorId = 0;
+                var waypoints = new List<Vector2>();
+                switch (pos)
                 {
-                    Minion m = new Minion(MinionSpawnType.MINION_TYPE_MELEE, positions[i], _laneWaypoints[i]);
-                    AddObject(m);
+                    case MinionSpawnPosition.SPAWN_BLUE_BOT:
+                        waypoints = _blueBotWaypoints;
+                        inhibitorId = 0xff26ac0f;
+                        break;
+                    case MinionSpawnPosition.SPAWN_BLUE_MID:
+                        waypoints = _blueMidWaypoints;
+                        inhibitorId = 0xffff8f1f;
+                        break;
+                    case MinionSpawnPosition.SPAWN_BLUE_TOP:
+                        waypoints = _blueTopWaypoints;
+                        inhibitorId = 0xff6793d0;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_BOT:
+                        waypoints = _redBotWaypoints;
+                        inhibitorId = 0xff9303e1;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_MID:
+                        waypoints = _redMidWaypoints;
+                        inhibitorId = 0xff4a20f1;
+                        break;
+                    case MinionSpawnPosition.SPAWN_RED_TOP:
+                        waypoints = _redTopWaypoints;
+                        inhibitorId = 0xffd23c3e;
+                        break;
                 }
-                return false;
-            }
+                var inhibitorIsDead = (GetInhibitorById(inhibitorId).getState() == InhibitorState.Dead);
 
-            if (_waveNumber == 3)
-            {
-                for (var i = 0; i < positions.Count; ++i)
+                if (_gameTime > 35 * 60 * 1000)
                 {
-                    Minion m = new Minion(MinionSpawnType.MINION_TYPE_CANNON, positions[i], _laneWaypoints[i]);
-                    AddObject(m);
+                    var list = inhibitorIsDead ? _superCannonMinionWave : _cannonMinionWave;
+                    SpawnMinion(list, _minionNumber, pos, waypoints);
                 }
-                return false;
-            }
+                else if (_gameTime > 20 * 60 * 1000)
+                {
+                    var list = inhibitorIsDead ?
```

Comment:
I'd do this whole section like this
```CS
var wavesToSuperWaves = new Dictionary<List<string>, List<string>>()
{
    {_regularMinionWave, _superMinionWave},
    {_cannonMinionWave, _superCannonMinionWave}
}

var cannonMinionTimestamps = new List<Tuple<long, int>>
{
    new Tuple<long, int>(0, 2),
    new Tuple<long, int>(20 * 60 * 100, 1),
    new Tuple<long, int>(35 * 60 * 100, 0)
}

var cannonMinionCap = 2;
foreach(var timestamp in cannonMinionTimestamps)
{
    if(_gameTime >= timestamp.Item1)
    {
        cannonMinionCap = timestamp.Item2;
    }
}

var list = _regularMinionWave;
if(_cannonMinionCount >= cannonMinionCap)
{
    list = _cannonMinionWave;
}
if(inhibitorIsDead)
{
    list = wavesToSuperWaves[list];
}
```
the `cannonMinionCap` can be then carried over to the latter part of the function in which I suggested the same change.